src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/DotNetXamarinProject.cs

```diff
@@ -94,7 +94,16 @@ public void AddSources (params string [] sources)
 
 		public virtual ProjectRootElement Construct ()
 		{
+			// Workaround for https://github.com/dotnet/msbuild/issues/2554 when using Microsoft.Build.Construction.ProjectRootElement.Create
+			var msbuildExePathVarName = "MSBUILD_EXE_PATH";
+			if (!Builder.UseDotNet && !TestEnvironment.IsWindows) {
+				Environment.SetEnvironmentVariable (msbuildExePathVarName, typeof (DotNetXamarinProject).Assembly.Location);
+			}
 			var root = ProjectRootElement.Create ();
+			if (!Builder.UseDotNet && !TestEnvironment.IsWindows) {
+				Environment.SetEnvironmentVariable (msbuildExePathVarName, null);
+			}
```

Comment:
The only thing here might be to use a `try-finally` and clear the env var in the `finally` block.

Otherwise LGTM.

Reply:
Updated to use try/finally.